### ISSUE TITLE
[Grid - Grouping] All buttons that expands groups have the same name

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3387]
### 😯 Current Behavior

For each toggle button that is used to expand/collapse groups in Grid - Grouping stories the name is the same "Toggle group". 
This is not sufficient because it's not allowing screen reader user to get full information about group that can be expanded using each button. It's best shown when user list all the buttons in grid (for VoiceOver `control + option + U`) and then list of identical names appears.

https://github.com/user-attachments/assets/67f11a10-f9a5-42a8-9669-bc931a2f92ee

### 🤔 Expected Behavior/possible solution?

Name of each button should be unique and should allow user to easily tell what group exactly will be toggled with each button.
Each name should look like: 'Toggle Group NAME OF GROUP' or 'NAME OF GROUP toggle group'. 
Names can be added with `aria-labelledby` or `aria-label`. `aria-labelledby` is the most recommended solution in this case.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/grid-grouping--basic
2. With screen reader turned on place your focus on each button that expand each group
3. Check if every button has name that makes it easy to tell what's the button for

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/grid-grouping--basic

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Other, All

### Component version

v0.7.2

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
